pkg/agents/team6/config.go

```diff
@@ -0,0 +1,113 @@
+package team6
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/abm"
+	"github.com/SOMAS2021/SOMAS2021/pkg/messages"
+)
+
+type behaviour float64
+
+// const (
+//  altruist behaviour = iota
+//  collectivist
+//  selfish
+//  narcissist
+// )
+
+type team6Config struct {
+	baseBehaviour behaviour
+	//the scaling factor which limits the change in agent behaviour
+	stubbornness float64
+	//the largest jump in behaviour an agent can take
+	maxBehaviourSwing float64
+	//weights used to assess score for behaviour update
+	paramWeights behaviourParameterWeights
+	//floor scaling discount factor
+	lambda float64
+	//maximum behaviour score an agent can reach
+	maxBehaviourThreshold behaviour
+}
+
+type CustomAgent6 struct {
+	*infra.Base
+	config team6Config
+	//keep track of the lowest floor we've been to
+	maxFloorGuess int
+	currBehaviour behaviour
+}
+
+type thresholdBehaviourPair struct {
+	threshold behaviour
+	bType     string
+}
+
+type behaviourParameterWeights []float64
+
+var maxBehaviourThreshold behaviour = 10.0
+
+func chooseInitialBehaviour() behaviour {
+	rand.Seed(time.Now().UnixNano())
+	return behaviour(rand.Float64()) * maxBehaviourThreshold
+}
+
+func New(baseAgent *infra.Base) (abm.Agent, error) {
+	initialBehaviour := chooseInitialBehaviour()
+	return &CustomAgent6{
+		Base: baseAgent,
+		config: team6Config{
+			baseBehaviour:         initialBehaviour,
+			stubbornness:          0.0,
+			maxBehaviourSwing:     8,
+			paramWeights:          behaviourParameterWeights{0.7, 0.3}, //ensure sum of weights = max behaviour enum
+			lambda:                3.0,
+			maxBehaviourThreshold: maxBehaviourThreshold,
+		},
+		currBehaviour: initialBehaviour,
+		maxFloorGuess: baseAgent.Floor() + 2,
+	}, nil
+}
+
+func (b behaviour) String() string {
+
+	behaviourMap := [...]thresholdBehaviourPair{{2, "Altruist"}, {7, "Collectivist"}, {9, "Selfish"}, {10, "Narcissist"}}
+
+	if b >= 0 {
+
+		for _, v := range behaviourMap {
+			if b <= v.threshold {
+				return v.bType
+			}
+		}
+	}
+
+	return fmt.Sprintf("UNKNOWN Behaviour '%v'", int(b))
+}
+
+func (a *CustomAgent6) Run() {
+	log.Printf("Custom agent %s", a.ID())
```

Comment:
there is a new `a.Log()` function
- it has by default the ID and type of the agent printed
- on the top level we redirect where the logs go
- outputs in json format
- you can append fields to it
would be great if you can use the `a.Log()` instead 😄